### DISCUSSION TITLE
grafana集成milvus2.1没有数据

### DISCUSSION THREAD

**Discussion by Anandonzy** [milvus-io/milvus#18446] [Q&A and General discussion]
Grafana8.5.0 and grafana 5.X
Milvus 2.1.0
Prometheus 2.5.0
Milvus 使用Docker-compose 搭建的
在docker-compose 里面配置了node_exporter 的配置
`services:
  node-exporter:
    image: prom/node-exporter:latest
    hostname: node-exporter
    container_name: node-exporter
    ports:
      - "9100:9100"
    depends_on:
      - prometheus
    networks:
      - prom`
      我们有两个grfana 之前询问的可能是版本的问题然后想在新的8.5.0版本里面监控你们提供的模板的指标数据
      `https://github.com/milvus-io/milvus/tree/master/deployments/monitor/grafana`
      导入之后发现老版本的grafana里面展示的已经有node_exporter 信息了.
      但是新版本没有你们提供的模板指标信息.
      问题:1.想问下是不是我的docker-compose 里面少配置了监控指标信息的配置
      问题详情见图1(新grafana):
<img width="1422" alt="image" src="https://user-images.githubusercontent.com/32696359/181502027-176bbfe6-d4e8-49bf-97e4-c3274e4bbd95.png">
node_exporter (老grafana可以展示)展示见图2:
<img width="1256" alt="image" src="https://user-images.githubusercontent.com/32696359/181502161-d45ca166-f2d6-4146-adcc-719fbf20a3dd.png">
感谢回复~

**Comment by LoveEachDay** [comment]
@Anandonzy You cannot use node-exporter to scrape milvus metrics. 
You need to add a specific scrape target for prometheus. By the way, milvus components expose their metrics at `:9091/metrics`.

Another thing to note, you need to change the variable settings for the `milvus-dashboard` . By default, we use `kube_pod_info` metric to fetch the `namespace` and `go_info` metrics to fetch milvus instance. You need write your own dashboard according to the metrics doc at [here](https://github.com/milvus-io/milvus/blob/master/deployments/monitor/grafana/README.md). 

You'd better deploy your milvus into a kubernetes cluster, and take a reference [here](https://milvus.io/docs/v2.1.x/monitor.md) to enjoy milvus observability.

**Reply by Eddie-0721** [reply]
如果是用docker-compose部署的milvus，请问应该怎么修改模板呢，另外是否可以提供下，prometheus的告警rule呢，感谢回复

**Reply by yhmo** [reply]
相关讨论，我们不提供docker-compose的deshboard： ttps://github.com/milvus-io/milvus/issues/32981
“If the instance is deployed through Docker, then we don't actually provide a corresponding monitoring dashboard.”

可自行创建告警，根据自己需求自行创建：https://milvus.io/docs/alert.md
